easybuild/framework/easyblock.py

```diff
@@ -1444,8 +1443,19 @@ def extensions_step(self, fetch=False):
         self.clean_up_fake_module(fake_mod_data)
 
     def package_step(self):
-        """Package software (e.g. into an RPM)."""
-        pass
+        """Prepare package software (e.g. into an RPM) with fpm."""
+        rpmname = "HPCBIOS.20150211-%s-%s" % (self.name, self.version)
+        os.chdir(os.environ['TMPDIR'])
+        
+        if self.toolchain.name == "dummy":
+            dependencies = []
+        else:
+            dependencies = [ "=".join([ self.toolchain.name, self.toolchain.version ]) ]
+        dependencies.extend([ "=".join([ dep['name'], dep['version'] ]) for dep in self.cfg.dependencies() ])
+        depstring = '--dependency ' + ' --dependency '.join(dependencies)
+
+        cmd = "fpm --workdir $TMPDIR -t rpm --name %s -s dir %s -C %s" % (rpmname, depstring, self.installdir)
```

Comment:
If it would be useful, I would be open to exploring some kind of rpc
interface (jsonrpc?) to fpm instead of requiring programs invoke the cli
with flags. Not required for this PR, but just putting the idea out there :)

On Sunday, February 22, 2015, Stijn De Weirdt <notifications@github.com>
wrote:

> In easybuild/framework/easyblock.py
> <https://github.com/hpcugent/easybuild-framework/pull/1170#discussion_r25146407>
> :
>
> > @@ -1444,8 +1443,19 @@ def extensions_step(self, fetch=False):
> >          self.clean_up_fake_module(fake_mod_data)
> >
> >      def package_step(self):
> > -        """Package software (e.g. into an RPM)."""
> > -        pass
> > +        """Prepare package software (e.g. into an RPM) with fpm."""
> > +        rpmname = "HPCBIOS.20150211-%s-%s" % (self.name, self.version)
> > +        os.chdir(os.environ['TMPDIR'])
> > +
> > +        if self.toolchain.name == "dummy":
> > +            dependencies = []
> > +        else:
> > +            dependencies = [ "=".join([ self.toolchain.name, self.toolchain.version ]) ]
> > +        dependencies.extend([ "=".join([ dep['name'], dep['version'] ]) for dep in self.cfg.dependencies() ])
> > +        depstring = '--dependency ' + ' --dependency '.join(dependencies)
> > +
> > +        cmd = "fpm --workdir $TMPDIR -t rpm --name %s -s dir %s -C %s" % (rpmname, depstring, self.installdir)
>
> @jordansissel <https://github.com/jordansissel> running everything in
> subshell was a historical mistake we made in EB long long ago. it's very
> convenient, maybe even required, for some cirtical parts of EB (like
> conifgure and make), but we shouldn't be using it were it can be easily
> avoided like in this case. so we should already try to write code that
> assumes we will move to non-subshell (as one of my earlier comments
> suggested).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hpcugent/easybuild-framework/pull/1170/files#r25146407>
> .
>